### ISSUE TITLE
runtime/race: race_linux_amd64.syso now depends on glibc 2.17

### ISSUE THREAD

**Issue by djedward** [golang/go#53522]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.18.3 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes -> 1.19beta1

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GOARCH="amd64"
GOBIN="/workspace/src/GoLang/build/bin"
GOCACHE="/workspace/src/GoLang/build/private/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH=""
GOPROXY=""
GORACE=""
GOROOT="/workspace/src/GoLang/build/lib"
GOTMPDIR=""
GOTOOLDIR="/workspace/src/GoLang/build/lib/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build226727772=/tmp/go-build -gno-record-gcc-switches" 
</pre></details>

### What did you do?

I was attempting to build go1.19beta1 on one of our older Linux distributions which has glibc 2.12. go1.18.3 builds work fine. It fails with:
<pre>
##### Testing race detector
# runtime/race
runtime/race/race_linux_amd64.syso:gotsan.cpp:function __sanitizer::MonotonicNanoTime(): error: undefined reference to 'clock_gettime'
collect2: error: ld returned 1 exit status
FAIL    runtime/race
FAIL
# runtime/race
runtime/race/race_linux_amd64.syso:gotsan.cpp:function __sanitizer::MonotonicNanoTime(): error: undefined reference to 'clock_gettime'
collect2: error: ld returned 1 exit status
FAIL    flag
FAIL    net
FAIL    os
FAIL    os/exec
FAIL    encoding/gob
FAIL
# runtime/race
runtime/race/race_linux_amd64.syso:gotsan.cpp:function __sanitizer::MonotonicNanoTime(): error: undefined reference to 'clock_gettime'
collect2: error: ld returned 1 exit status
FAIL    flag
FAIL    os/exec
FAIL 
</pre>

Grabbing the previous version of <code>race_linux_amd64.syso</code> (prior to commit d6a1ffd) appears to allow the build to proceed. It seems this may be a recurrence of #37485. It appears to have a dependency on glibc 2.17 now, where as before it was glibc 2.12.

### What did you expect to see?

Successful build of 1.19beta1

### What did you see instead?

Failed race detector tests

**Comment by ianlancetaylor** [comment]
The current version of glibc is 2.35.  glibc 2.17 was released in 2012, almost 10 years ago.  I think this dependency is OK.  We have to be able to move forward sometimes.

**Comment by djedward** [comment]
I can see wanting to move forward as completely reasonable and it isn't a show-stopper for us. Just saw what might be an unintended regression in 1.19beta1 (and didn't catch anything in release notes indicating drop of support).

Though it is probably worth mentioning that the [Go Minimum Requirements](https://github.com/golang/go/wiki/MinimumRequirements) still lists kernel 2.6.23 as supported, which predates even glibc 2.12.

**Comment by ianlancetaylor** [comment]
Thanks for pointing that out.  We actually lifted the minimum Linux version to 2.6.32 in #45964, but we never updated the wiki page.  I just did that now.

Although 2.6.32 is still older than glibc 2.17.

And that said, Go still works with glibc 2.17, it's only the race detector that fails.  Hopefully people still using these old glibc versions can at least have access to a newer glibc if they want to run the race detector.
